src/app/core/services/api/fake-backend/connector-fake-impl/asset-fake-service.ts

```diff
@@ -17,34 +19,75 @@ export const assetPage = (): AssetPage => {
 export const getAssetById = (id: string) =>
   assets.find((it) => it.assetId === id);
 
+function patchAsset(assetId: string, patcher: Patcher<UiAsset>): UiAsset {
+  assets = assets.map((it) =>
+    it.assetId === assetId ? patchObj(it, patcher) : it,
+  );
+  return getAssetById(assetId)!;
+}
+
 export const createAsset = (asset: UiAssetCreateRequest): IdResponseDto => {
+  const assetId = asset.id;
   assets.push({
-    assetId: asset.id,
-    title: asset.title ?? asset.id,
+    assetId,
+    ...metadata(assetId, asset),
     connectorEndpoint: 'https://my-connector/api/dsp',
     participantId: 'MDSL1234XX.C1234XX',
-    description: asset.description,
     creatorOrganizationName: 'My Org',
-    publisherHomepage: asset.publisherHomepage,
-    licenseUrl: asset.licenseUrl,
-    version: asset.version,
-    keywords: asset.keywords,
-    mediaType: asset.mediaType,
-    landingPageUrl: asset.landingPageUrl,
-    dataCategory: asset.dataCategory,
-    dataSubcategory: asset.dataSubcategory,
-    dataModel: asset.dataModel,
-    geoReferenceMethod: asset.geoReferenceMethod,
-    transportMode: asset.transportMode,
-    additionalProperties: asset.additionalProperties,
-    privateProperties: asset.privateProperties,
   });
   return {
     id: asset.id,
     lastUpdatedDate: new Date(),
   };
 };
 
+export const editAssetMetadata = (
+  assetId: string,
+  request: UiAssetEditMetadataRequest,
+): IdResponseDto => {
+  const asset = patchAsset(assetId, () => metadata(assetId, request));
+
+  return {
+    id: asset.assetId,
+    lastUpdatedDate: new Date(),
+  };
+};
+
+function metadata(
+  assetId: string,
+  request: UiAssetCreateRequest | UiAssetEditMetadataRequest,
+): Omit<
+  UiAsset,
+  | 'assetId'
+  | 'assetJsonLd'
+  | 'connectorEndpoint'
+  | 'creatorOrganizationName'
```

Comment:
Is it maybe worth it to create a new type for this as these seem to be the editable fields of the UiAsset?

Reply:
i would prefer not having to give this thing a name. it is just there to help me DRY between the two functions that are "somewhat" similar, while staying 100% type-safe